### ISSUE TITLE
pytest fails on local supercomputer (Skylake cpus)

### ISSUE THREAD

**Issue by manodeep** [manodeep/Corrfunc#283]
## General information

- Corrfunc version:2.4.0
- platform:linux
- installation method (pip/source/other?):git

## Issue description 
Cloned a fresh repo on local supercomputer with Skylake cpus and installed with `python -m pip install -e .` and then ran `pytest -v`

### Expected behavior 
Tests should pass

### Actual behavior
Most tests failed, there were only a handful of tests that seemed to pass

### What have you tried so far?
Unsure what is going on. GH Actions on similar AVX512F cpus (may be skylake) seems to pass just fine

**Comment by lgarrison** [comment]
This is on `master` branch, right? What's the failure mode of the tests---wrong pair counts, wrong weights, crashing, etc? Is it worth sharing the test output in case there's a pattern to which ones fail?

**Comment by manodeep** [comment]
Good point! Here is the log from before (not from the fresh one, but should be the same)
[Corrfunc_master_pytest.log](https://github.com/manodeep/Corrfunc/files/9943913/Corrfunc_master_pytest.log)

**Comment by lgarrison** [comment]
None of the failures are single-threaded, so it's probably an OpenMP library issue. Maybe the same one from here? https://github.com/manodeep/Corrfunc/pull/258#issuecomment-966474906
